hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -171,6 +168,22 @@ public long takeSnapshot() throws IOException {
     return 0;
   }
 
+  /**
+   * Leader OM has purged entries from its log. To catch up, OM must download
+   * the latest checkpoint from the leader OM and install it.
+   * @param firstTermIndexInLog TermIndex of the first append entry available
+   *                           in the Leader's log.
+   * @return the last term index included in the installed snapshot.
+   */
+  public CompletableFuture<TermIndex> notifyInstallSnapshotFromLeader(
+      TermIndex firstTermIndexInLog) {
+    // TODO: Raft server should send the leaderId
+    String leaderId = null;
+    CompletableFuture<TermIndex> future = CompletableFuture
+        .supplyAsync(() -> ozoneManager.installSnapshot(leaderId));
```

Comment:
We should not execute this in the default ForkJoinPool. That can suffer from thread exhaustion/deadlock issues since there are very few threads in the default pool.

Instead use the overload of `supplyAsync` that accepts an `Executor`.

Reply:
Done